be/src/olap/memtable_flush_executor.cpp

```diff
@@ -132,10 +132,12 @@ void MemTableFlushExecutor::init(const std::vector<DataDir*>& data_dirs) {
 }
 
 // NOTE: we use SERIAL mode here to ensure all mem-tables from one tablet are flushed in order.
+// unique key merge on write should flush serial cause calc delete bitmap should load segment serial
 Status MemTableFlushExecutor::create_flush_token(std::unique_ptr<FlushToken>* flush_token,
-                                                 RowsetTypePB rowset_type, bool is_high_priority) {
+                                                 RowsetTypePB rowset_type, bool merge_on_write,
+                                                 bool is_high_priority) {
     if (!is_high_priority) {
-        if (rowset_type == BETA_ROWSET) {
+        if (rowset_type == BETA_ROWSET && !merge_on_write) {
```

Comment:
L150 should also consider the parameter.

Reply:
done